crates/trie/trie/src/updates.rs

```diff
@@ -40,6 +40,79 @@ impl TrieOp {
     }
 }
 
+/// `StorageWriter` is responsible for writing trie updates to the database.
+///
+/// This struct encapsulates the logic required to flush trie operations to the appropriate
+/// database tables.
+#[derive(Debug, Default)]
+pub struct StorageWriter;
```

Comment:
I'd like to move this to the provider crate for now

so that we no longer need to depend on any db stuff in trie

Reply:
Fixed, but it seems we have a very strange `the crate `reth_trie` is compiled multiple times, possibly with different configurations` error not linked to this PR no? Seems to be linked with the `updates` module in trie wdyt?